test/socket.jl

```diff
@@ -253,16 +253,27 @@ let
         bind(s, ip"0.0.0.0", 2000, reuseaddr = true, enable_broadcast = true)
         s
     end
+
+    function wait_until_timeout(recvs)
+        TIMEOUT_VAL = 3  # seconds
+        t0 = time()
+        recvs_check = copy(recvs)
+        while ((length(filter!(t->!istaskdone(t), recvs_check)) > 0)
+              && (time() - t0 < TIMEOUT_VAL))
+            sleep(0.05)
+        end
+        length(recvs_check) > 0 && error("timeout")
+        map(wait, recvs)
```

Comment:
Or not, since the `wait` will re-raise any errors from the async tasks. Ignore above comment.